[CI] Unable to parse policy file

**Build Scans:**
- [elasticsearch-periodic #6314 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/lc6udysmp7wmc)
- [elasticsearch-periodic #6307 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/ixwpr3prvzbrs)
- [elasticsearch-periodic #6293 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/acgsa5q7tdrs4)
- [elasticsearch-periodic #6279 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/7fupodx2r52kq)
- [elasticsearch-periodic #6272 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/mgdsih4bp6wvm)
- [elasticsearch-periodic #6244 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/amhqncsowgkki)
- [elasticsearch-periodic #6237 / openjdk24_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/x2yox4nvj3sr4)

**Reproduction Line:**
```
undefined
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.qa.AsyncSearchHeadersIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.qa.AsyncSearchHeadersIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
undefined
```

**Issue Reasons:**
- [main] 7 consecutive failures in step openjdk24_checkpart2_java-matrix
- [main] 7 failures in class org.elasticsearch.qa.AsyncSearchHeadersIT (1.5% fail rate in 462 executions)
- [main] 7 failures in step openjdk24_checkpart2_java-matrix (100.0% fail rate in 7 executions)
- [main] 7 failures in pipeline elasticsearch-periodic (58.3% fail rate in 12 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Just clicking through to the logs here since there is no failure message in the issue:

```
[2025-02-10T13:35:45,707][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] java.lang.IllegalArgumentException: unable to parse policy file `file:/dev/shm/bk/bk-agent-prod-gcp-1739192775415162351/elastic/elasticsearch-periodic/x-pack/plugin/async-search/qa/rest/build/testrun/javaRestTest/temp/test-cluster14751187041045127251/test-cluster-0/distro/plugins/.installing-5492501003814694756/plugin-security.policy`
[2025-02-10T13:35:45,707][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.bootstrap.PolicyUtil.readPolicy(PolicyUtil.java:307)
[2025-02-10T13:35:45,708][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.bootstrap.PolicyUtil.readPolicyInfo(PolicyUtil.java:344)
[2025-02-10T13:35:45,708][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.bootstrap.PolicyUtil.getPluginPolicyInfo(PolicyUtil.java:381)
[2025-02-10T13:35:45,708][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginAction.installPlugin(InstallPluginAction.java:925)
[2025-02-10T13:35:45,708][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginAction.execute(InstallPluginAction.java:255)
[2025-02-10T13:35:45,708][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginCommand.execute(InstallPluginCommand.java:90)
[2025-02-10T13:35:45,709][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.common.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:55)
[2025-02-10T13:35:45,710][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:95)
[2025-02-10T13:35:45,710][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:95)
[2025-02-10T13:35:45,710][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:95)
[2025-02-10T13:35:45,711][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.cli.Command.main(Command.java:52)
[2025-02-10T13:35:45,711][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.launcher.CliToolLauncher.main(CliToolLauncher.java:65)
[2025-02-10T13:35:45,711][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] Caused by: java.security.NoSuchAlgorithmException: JavaPolicy Policy not available
[2025-02-10T13:35:45,711][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:189)
[2025-02-10T13:35:45,712][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at java.base/java.security.Policy.getInstance(Policy.java:162)
[2025-02-10T13:35:45,712][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.bootstrap.PolicyUtil.readPolicy(PolicyUtil.java:290)
[2025-02-10T13:35:45,712][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	... 11 more
```

Another test failure that shows up in the build scans looks like https://github.com/elastic/elasticsearch/issues/121967

--- a/distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/cli/InstallPluginAction.java
+++ b/distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/cli/InstallPluginAction.java
@@ -38,6 +38,7 @@ import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.core.Tuple;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.jdk.JarHell;
+import org.elasticsearch.jdk.RuntimeVersionFeature;
 import org.elasticsearch.plugin.scanner.ClassReaders;
 import org.elasticsearch.plugin.scanner.NamedComponentScanner;
 import org.elasticsearch.plugins.Platforms;
@@ -922,10 +923,12 @@ public class InstallPluginAction implements Closeable {
      */
     private PluginDescriptor installPlugin(InstallablePlugin descriptor, Path tmpRoot, List<Path> deleteOnFailure) throws Exception {
         final PluginDescriptor info = loadPluginInfo(tmpRoot);
-        PluginPolicyInfo pluginPolicy = PolicyUtil.getPluginPolicyInfo(tmpRoot, env.tmpDir());
-        if (pluginPolicy != null) {
-            Set<String> permissions = PluginSecurity.getPermissionDescriptions(pluginPolicy, env.tmpDir());
-            PluginSecurity.confirmPolicyExceptions(terminal, permissions, batch);
+        if (RuntimeVersionFeature.isSecurityManagerAvailable()) {
+            PluginPolicyInfo pluginPolicy = PolicyUtil.getPluginPolicyInfo(tmpRoot, env.tmpDir());
+            if (pluginPolicy != null) {
+                Set<String> permissions = PluginSecurity.getPermissionDescriptions(pluginPolicy, env.tmpDir());
+                PluginSecurity.confirmPolicyExceptions(terminal, permissions, batch);
+            }
         }
 
         // Validate that the downloaded plugin's ID matches what we expect from the descriptor. The

